batchspawner/singleuser.py

```diff
@@ -34,6 +36,16 @@ def main(argv=None):
         **kwargs,
     )
 
+    # Change the port number for the environment variable JUPYTERHUB_SERVICE_URL
+    url = urlparse(os.environ["JUPYTERHUB_SERVICE_URL"])
+    url_netloc = url.netloc.split(":")
+
+    if len(url_netloc) == 2:
```

Comment:
This method of setting the port would lead to the new port not being set if the netloc in the original JUPYTERHUB_SERVICE_URL doesn't have the port explicitly defined (e.g. https://example.com/test wouldn't change). Meanwhile the solution in PR #267 for this issue does replace the port in all cases.